packages/vitest/src/runtime/runners/benchmark.ts

```diff
@@ -165,9 +165,10 @@ export class NodeBenchmarkRunner implements VitestRunner {
     return await import('tinybench')
   }
 
-  importFile(filepath: string, source: VitestRunnerImportSource): unknown {
+  async importFile(filepath: string, source: VitestRunnerImportSource): Promise<unknown> {
     if (source === 'setup') {
-      getWorkerState().moduleCache.delete(filepath)
+      const resolved = await this.__vitest_executor.resolveUrl(filepath)
```

Comment:
Shouldn't it be resolved here even without the extension? https://github.com/vitest-dev/vitest/blob/63f8b07042e9885d74daca7e175351c47078b99a/packages/vitest/src/node/config/resolveConfig.ts#L311

This worked before without problems

Reply:
I imagine it's working for most of the case, but the issue was when it's extension-less relative path. For example, `setupFiles: ['./setup.ts']` works but `setupFiles: ['./setup']` fails, which was OP's repro.

Alternatively, we can use Vite server resolver during `resolveConfig` or sometime later if we restructure a bit, but I thought there might be a reason not to bring that.

Reply:
Yeah, we should fix it for sure, but I don't think this was a regression and extension-less relative path has never been handled with proper invalidation. I'll think about the solution.

Reply:
In any way, I don't think the current fix in this PR is the correct one. In the very least it seems weird to me that it processes the same setup file in every test (which means it processes the same value multiple times). The setup file ID should be resolved already at the config stage. The `resolvePath` in `resolveConfig` is supposed to be able to resolve any JS paths even without an extension as far as I understand - why doesn't it? 🤔